.github/workflows/build.yml

```
@@ -0,0 +1,142 @@
+name: Build Pulsar Binaries
+
+on:
+  pull_request:
+  push:
+    branches: [ 'master' ]
+
+env:
+  # Variables needed for build information
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  PYTHON_VERSION: '3.10'
+  NODE_VERSION: 16
+  # Below variables allow us to quickly control visual tests for each platform
+  RUN_WINDOWS_VT: false
+  RUN_LINUX_VT: true
+  RUN_MACOS_VT: false
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest, macos-latest ]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - name: Checkout the latest code
+      uses: actions/checkout@v3
+
+    - name: Setup Node
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ env.NODE_VERSION }}
+
+    - name: Setup Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ env.PYTHON_VERSION }}
+
+    - name: Setup Git Submodule
+      run: |
+        git submodule init
+        git submodule update
+
+    - name: Check Pulsar Version
+      if: ${{ matrix.os != 'windows-latest' }}
+      run: sed -i -e "s/[0-9]*-dev/`date -u +%Y%m%d%H`/g" package.json
+
+    - name: Check Pulsar Version - Windows
+      if: ${{ matrix.os == 'windows-latest' }}
+      run: (Get-Content package.json) -replace '[0-9]*-dev', (date -u +%Y%m%d%H) | Set-Content -Path package.json
+
+    - name: Install Pulsar Dependencies
+      uses: nick-fields/retry@v2
+      with:
+        timeout_minutes: 30
+        max_attempts: 3
+        retry_on: error
+        command: yarn install --ignore-engines
+        on_retry_command: rm -R node_modules
+
+    - name: Build Pulsar
+      uses: nick-fields/retry@v2
+      with:
+        timeout_minutes: 30
+        max_attempts: 3
+        retry_on: error
+        command: |
+          yarn build
+          yarn run build:apm
+
+    # macOS Signing Stuff
+    - name: Build Pulsar Binaries (macOS)
+      if: ${{ matrix.os == 'macos-latest' }}
```

Broader macOS check, like the windows one above

https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context


> `runner.os`: The operating system of the runner executing the job. Possible values are `Linux`, `Windows`, or `macOS`.

```suggestion
      if: ${{ runner.os == 'macOS' }}
```